drivers/soc/bl702/std/startup/system_bl702.c

```diff
@@ -57,13 +57,13 @@ void SystemInit(void)
     BL_WR_REG(GLB_BASE, GLB_PARM, tmpVal);
 
     /* CLear all interrupt */
-    p = (uint32_t *)(CLIC_HART0_BASE + CLIC_INTIE_OFFSET);
+    p = (uint8_t *)(CLIC_HART0_BASE + CLIC_INTIE_OFFSET);
 
     for (i = 0; i < (IRQn_LAST + 3) / 4; i++) {
         p[i] = 0;
     }
 
-    p = (uint32_t *)(CLIC_HART0_BASE + CLIC_INTIP_OFFSET);
+    p = (uint8_t *)(CLIC_HART0_BASE + CLIC_INTIP_OFFSET);
```

Comment:
uint8_t is incorrect because with `for (i = 0; i < (IRQn_LAST + 3) / 4; i++) `

Reply:
From e24 RISC-V core description:
![изображение](https://github.com/bouffalolab/bouffalo_sdk/assets/46779568/f1ffba7f-8599-4974-ac82-0bf73098a806)
![изображение](https://github.com/bouffalolab/bouffalo_sdk/assets/46779568/5f711d02-0865-420d-b11b-5a85becc59e1)
registers are 1B width and address is computed as `CLIC Hart Base + 1 × Interrupt ID`, but as I see now the `/4` devision should be removed. So cycle should be `for (i = 0; i < IRQn_LAST; i++)`

Reply:
Use uint32_t to make it quickly.

Reply:
uint32_t usage is incorrect, cause it will produce 4B aligned addresses while accessing pointer as array. CLIC registers are **byte aligned**, not **word aligned** and so their pointer type should be `uint8_t *`. Please look at provided figures.

Reply:
@mdednev CLIC_INTIE_OFFSET and CLIC_INTIP_OFFSET are both 4 byte aligned. the number of registers is evenly divisible by 4. they're all being set to 0 anyway.
As long as the cpu supports writing 32bit values here the original code should do the correct thing, right?
Is there something else here that we're not seeing?

Reply:
> @mdednev CLIC_INTIE_OFFSET and CLIC_INTIP_OFFSET are both 4 byte aligned.

Yes, this is right.

> the number of registers is evenly divisible by 4. 

It depends on implementation and it isn't true in general. In other words this is error-prone way. So I've suggested more general and error-proof initialization code. Yes, it is little-bit slower, but it is more correct and matches e24 core registers description (1B registers should be accessed as 1B and not as 4B). For example SDK IRQ driver uses **1B** accesses to this registers (drivers/lhal/src/bflb_irq.c):
```C
void bflb_irq_disable(int irq)
{
#if defined(BL702) || defined(BL602) || defined(BL702L)
    putreg8(0, CLIC_HART0_BASE + CLIC_INTIE_OFFSET + irq);
#else
    csi_vic_disable_irq(irq);
#endif
}
```

> they're all being set to 0 anyway. As long as the cpu supports writing 32bit values here the original code should do the correct thing, right? Is there something else here that we're not seeing?

Yes, this code works, but if it will be reused for future MCUs it could lead to errors in initialization.